src/smc-webapp/latex/editor.coffee

```diff
@@ -513,16 +515,20 @@ class exports.LatexEditor extends editor.FileEditor
 
     # This function isn't called on save
     # @latex_editor.save is called instead for some reason
-    save: (cb, force=false) =>
+    save: (cb, force) =>
+        force ?= false
         @latex_editor.save (err) =>
             cb?(err)
             if not err
-                @update_preview (force=force) =>
+                sf =  =>
                     if @_current_page == 'pdf-preview'
                         @preview_embed?.update()
+                @update_preview(sf, force)
                 @spell_check()
 
-    update_preview: (cb, force=false, only_compile=false) =>
+    update_preview: (cb, force, only_compile) =>
```

Comment:
for what it is worth:

(1) I think what you had originally does work fine in CS.  What doesn't work is calling it like `@update_preview(cb, only_compile=true)`.

(2) I would GREATLY prefer is you use defaults/opts here.  It's used about 10K other places in our codebase, solves precisely this problem, leads to very clear code, etc. Please use that rather than `force ?=` etc.  It's easy.